src/main/php/PHPMD/Renderer/RendererFactory.php

```diff
@@ -14,4 +15,47 @@ public static function createBaselineRenderer(AbstractWriter $writer): BaselineR
 
         return $renderer;
     }
+
+    /**
+     * @throws InvalidArgumentException
+     */
+    public function getRenderer(string $format): RendererInterface
```

Comment:
The old code handled null via default, we should probably keep that behavior or it will require some additional refactoring to avoid null being possible.

```suggestion
    public function getRenderer(?string $format): RendererInterface
```

Reply:
I prefer to keep signature un-changed, because format should be a required string parameter only. So with commit https://github.com/phpmd/phpmd/pull/1197/commits/c9f26b3537d067262b8257977a643600fe22608b I've replaced [Elvis operator](https://en.wikipedia.org/wiki/Elvis_operator) syntax by [Null coalescing](https://en.wikipedia.org/wiki/Null_coalescing_operator#PHP) syntax.

Reply:
null coalescing will ignore a missing property, so maybe elvis was actually the better choice.

Reply:
I don't want to change signature of `getRenderer` and `getCustomRenderer` just because `$this->reportFormat` is possibly undefined. 
With `$reportFormat = $reportFormat ?? $this->reportFormat ?? '';` we have always a default (string) value and not a null value that the `class_exists` will not handle correctly

Reply:
I'm not telling you to change the signature, I'm saying that it's better to use Elvis then null coralsesing to check for a nullable property since it doesn't hide undefined properties.

Reply:
Yes I saw it before writing. It does not address my comment.

Reply:
I'm sorry but I don't understand what you want. If you want to change something please provide a patch. It will be more clear

Reply:
https://github.com/phpmd/phpmd/pull/1197#discussion_r1686547605

Reply:
Done : https://github.com/phpmd/phpmd/pull/1197/commits/71283feb5276f845dbf124b3ce7bd11fbeac199a